### PR TITLE
prowgen: generate kvm label

### PR DESCRIPTION
**Pull request by hongkailiu**
https://issues.redhat.com/browse/DPTP-2017

/cc @openshift/openshift-team-developer-productivity-test-platform @cgwalters

### CODE REVIEW

File: pkg/prowgen/prowgen.go [hunk 1]
```diff
@@ -178,7 +178,12 @@ func GenerateJobs(configSpec *cioperatorapi.ReleaseBuildConfiguration, info *Pro
 			postsubmit.MaxConcurrency = 1
 			postsubmits[orgrepo] = append(postsubmits[orgrepo], *postsubmit)
 		} else {
-			presubmits[orgrepo] = append(presubmits[orgrepo], *generatePresubmitForTest(element.As, info, podSpec, configSpec.CanonicalGoRepository, jobRelease, skipCloning))
+			presubmit := *generatePresubmitForTest(element.As, info, podSpec, configSpec.CanonicalGoRepository, jobRelease, skipCloning)
+			v, requestingKVM := configSpec.Resources.RequirementsForStep(element.As).Requests["devices.kubevirt.io/kvm"]
```
Comment (alvaroaleman) [inline pkg/prowgen/prowgen.go:182]: Add a test, please

Comment (hongkailiu) [reply pkg/prowgen/prowgen.go]: Added.

File: pkg/prowgen/prowgen_test.go [hunk 2]
```diff
@@ -469,6 +469,21 @@ func TestGenerateJobs(t *testing.T) {
 				Repo:   "repository",
 				Branch: "branch",
 			}},
+		}, {
+			id: "kvm label",
+			config: &ciop.ReleaseBuildConfiguration{
+				Resources: map[string]ciop.ResourceRequirements{
+					"*": {Requests: ciop.ResourceList{"devices.kubevirt.io/kvm": "1"}},
```
Comment (stevekuznetsov) [inline pkg/prowgen/prowgen_test.go:476]: very clever, or too clever? only time will tell

**Review by alvaroaleman — APPROVED** [review]

File: pkg/prowgen/prowgen_test.go (hunk 2 above)
Comment (hongkailiu) [reply pkg/prowgen/prowgen_test.go:476]: Not sure if I get your point.
That is how our user uses it for now.

https://github.com/openshift/release/blob/0f651ff1091687662a4799e57f5455504a45a4d3/ci-operator/config/openshift/os/openshift-os-master.yaml#L13

I cannot see what it means to build pods though.
Is it what you think only time will tell?

Comment (hongkailiu) [reply pkg/prowgen/prowgen_test.go:476]: avoid to apply to build pods: the 2nd use case
https://github.com/openshift/release/blob/f896da9216d4742ce591f04fad80331dfc8f47c4/ci-operator/config/coreos/rpm-ostree/coreos-rpm-ostree-master.yaml#L26-L37

**Pull request merged**